pennylane/ops/qubit.py

```diff
@@ -167,11 +169,15 @@ class PauliZ(Observable, Operation):
     def _matrix(*params):
         return np.array([[1, 0], [0, -1]])
 
+    @staticmethod
+    def _diagonal(*params):
+        return np.array([1, -1])
```

Comment:
Just out of curiosity, could this functionality also be achieved by making eigvals an optional staticmethod of Operations, and not just Observables? Or is there a technical reason why the existing `eigvals` data can't be used?

Reply:
That's a good point! I'm also curious, by having the matrices hardcoded where known (an exception being e.g. `DiagonalQubitUnitary`), `_diagonal` would rather be defined for completeness, right?

Reply:
I initially was a little bit against using `eigvals`, because I was thinking of numpy's `eigvals` method _that returns eigenvalues in ascending order_. But for us, order matters! E.g. Z = diag(1, -1) but `np.eigvals(Z) = [-1, 1]`.

But if we just agree that our `eigvals` has the eigenvalues in computational basis order then we should be able to use it.

@antalszava, I'd actually say that for the `DiagonalOperation`, `_matrix` is just defined for completeness, because it could be `np.diag(_diagonal(...))` and is not used in the simulator.

Reply:
Worth double checking, but I believe that we already require eigvals to be in the computational basis ordering (hence the convenience pauli_eigs function)

Reply:
https://github.com/XanaduAI/pennylane/blob/56e321a2c3259854e8c70f4bb8d100bf33d360d6/pennylane/utils.py#L152

I agree that this is a fair point, it made a bit of confusion for myself as well. Therefore if the name seems to create the confusion, I'd be in for renaming/introducing another property.